### PR TITLE
Added GD32103C-START, GD32307C-EVAL, and GD32350C-START boards

### PR DESCRIPTION
**Pull request by howitzer74**
Added board support for the GD32103C-START, GD32307C-EVAL, and GD32350C-START development boards from Gigadevice.

### CODE REVIEW

File: boards/gd32307c_eval.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "0x0003"
+      ],
+      [
+        "0x1EAF",
+        "0x0004"
+      ]
+    ],
+    "mcu": "gd32f307vct6",
+    "product_line": "GD32F307xG",
+    "variant": "GD32F307VG_MBED",
```
Comment (maxgerhardt) [inline boards/gd32307c_eval.json:19]: Wait, if this is the board for the GD32F307VG_MBED variant then, then we need to delete https://github.com/CommunityGD32Cores/platform-gd32/blob/main/boards/gd32f307_mbed.json?  But the board file name is GD32307**C**, so is it really that VG board?

**Comment by maxgerhardt** [comment]
Also this PR adds the `GD32EPRT.svd` file which by its comments is 

>GD32E50x ARM 32-bit Cortex-M33 Microcontroller based 

but no other file references this file. (E.g., no reference in GD32E50x_CL). How is this file used?

File: boards/gd32307c_eval.json (hunk 1 above)
Comment (howitzer74) [reply boards/gd32307c_eval.json:19]: The GD32F307V-MBED is a different board from the GD32307C-EVAL board.  Note our interesting naming convention for these boards- sometimes it is GD32[EF]xxx while most of the time it is GD32xxx.  No rhyme or reason that I can tell!  We currently have 72 demo boards - I have attached a document listing them all for your reference.  I only have a few of these so I'm starting with those.
[GigadeviceBoardList_tabbed.txt](https://github.com/CommunityGD32Cores/platform-gd32/files/6903156/GigadeviceBoardList_tabbed.txt)

**Comment by howitzer74** [comment]
The GD32EPRT is a special variant of the GD32F50x that has  4MB PSRAM embedded in it, so this is the dedicated SVD file for it.

File: boards/gd32307c_eval.json (hunk 1 above)
Comment (maxgerhardt) [reply boards/gd32307c_eval.json]: resolved via 9f8b7d8a7f0a8fcb2d31054041e0c36cf05547f1

**Pull request merged**